### DISCUSSION TITLE
S3 failed to verify signature

### DISCUSSION THREAD

**Discussion by ivanpajon** [pocketbase/pocketbase#5635] [Q&A]
I'm deploying Pocketbase with Docker in Coolify (self-hosted).
I configure Pocketbase to connect IDrive E2 (S3 compatible) and it works fine, my endpoint url is like https://ID.idrivee2-50.com

Then I tried to use a custom CNAME configured in IDrive like https://idrivee2.mydomain.com and I'm getting this error:
```
"Failed to test the S3 filesystem. Raw error:
failed to upload a test file: blob (key "pb_settings_test_Ex65K/test.txt")
(code=Unknown): operation error S3: PutObject, https response error StatusCode: 403,
RequestID: XXX, HostID: XXX, api error SignatureDoesNotMatch: The request signature
we calculated does not match the signature you provided. Check your key and signing method."
```
![image](https://github.com/user-attachments/assets/1f37a993-31e8-4aa8-8cd1-fc5acecdf1b6)

It neither can't connect with "Force path-style addressing" and in my Coolify instance (https://coolify.mydomain.com) I have configured a S3 for backups with the custom CNAME endpoint and works fine.
I also have tried with latest [v0.23.0-rc3](https://github.com/pocketbase/pocketbase/releases/tag/v0.23.0-rc3) and I get the same error.

PD: I also tried with proposed solutions from https://github.com/pocketbase/pocketbase/discussions/801 and https://github.com/pocketbase/pocketbase/discussions/2019 with no result.

**Comment by ganigeorgiev** [comment]
I've just created a test "IDrive e2" account and it works correctly for me:

![idrive_screenshot](https://i.imgur.com/3Bhu9Lk.png)

Make sure that:
- that you use the correct endpoint, region and bucket name
- the issued access key have the correct permission (my test was with the defaults)
- that there are no typos or extra characters for the access key and secrets values

I don't have experience with Coolify so I can't help you with it, but try to reproduce the issue locally without docker, proxy or anything else in between, just PocketBase connecting directly to the S3 provider.

**Reply by ganigeorgiev** [reply]
Oh, on second read your issue seems to be related to your custom domain. In that case make sure that you have a valid TLS certificate for it, otherwise it will not work.

But in any case this is not really PocketBase related.

**Reply by ivanpajon** [reply]
Problem appear when you configure a custom CNAME in IDrive so in your case you could use something like https://custom.yourdomain.com as endpoint.

**Reply by ivanpajon** [reply]
I have a valid TLS certificate issued by Cloudflare, that is why I explained that in coolify the same configuration is working but here in pocketbase it doesn't...

**Reply by ganigeorgiev** [reply]
It doesn't make sense to work for the backups and not for the file upload configuration. Both uses the same implementation.

Again check your IDrive credentials and configuration but I don't think your problem is PocketBase related and this is not a general dev support forum.

**Reply by ivanpajon** [reply]
I opened this thread as a last resort, I didn't mean to be annoying, sorry. If I manage to fix it I will update with what worked for me in case it helps anyone else.

**Comment by gera2ld** [comment]
Hi I recently encountered this issue too. I finally figured out that it's because the `Accept-Encoding` header is altered by Cloudflare from `identity` to `gzip, br`.
I've explored possible solutions in #5917
